docs/developer/getting-started.rst

```diff
@@ -116,3 +116,13 @@ Building Documentation
 - Activate a conda environment with Mantid or ensure Mantid is in your ``PYTHONPATH``
 - If Mantid is unavailable (e.g. on Windows) edit ``docs/conf.py`` and include ``nbsphinx_allow_errors = True``. Take care to not commit this change though.
 - Run ``sphinx-build scipp/src/docs .``
+
+Precommit Hooks
+---------------
+
+If you wish, you can install precommit hooks for flake8 and yapf. In the source directory run:
+
+.. code-block:: bash
+
+  pre-commit install
+  pre-commit run --all-files
```

Comment:
Not sure how `pre-commit` works exactly. Our CI setup excludes certain files when running the formatting checks (see azure configs). How is this taken into account here?

Reply:
I'll check. If exclusions are in the tool's config file then they should be picked up, but that won't work if they are set using cmd line args in the CI scripts.

Reply:
That would be worth changing then, to be handled in the config files.

Reply:
I've restricted yapf to only run on files in the "python" directory to match CI.
There is no exclusion on flake8. CI config explicitly globs *.py, but that's redundant; flake8 does it by default.

Reply:
👍 Right, the exclusions I remembered were for `clang-format`.